src/services/__tests__/authenticationLockService.test.ts

```diff
@@ -66,3 +48,45 @@ describe("AuthenticationLockService#isUserAuthenticationLocked", () => {
     expect(result).toEqual(E.left(Error("an Error")));
   });
 });
+
+describe("AuthenticationLockService#lockUserAuthentication", () => {
+  const service = new AuthenticationLockService(lockedProfileTableClient);
+  const anUnlockCode = "anUnlockCode" as NonEmptyString;
+
+  it("should return true if CF-unlockcode has been stored sucessfully in table storage", async () => {
+    const result = await service.lockUserAuthentication(
+      aFiscalCode,
+      anUnlockCode
+    )();
+
+    expect(result).toEqual(E.right(true));
+  });
+
+  it("should return an Error when CF-unlockcode has already been stored in table storage", async () => {
```

Comment:
We have multiple options available:

1. Return a status code of 500 along with a ProblemJson containing a specific type. This will enable the front-end to identify the type and generate a new code.
2. Return a distinct status code to facilitate error differentiation for the front-end, which can then generate a new code.
3. In the front-end code, wait for all retry attempts to fail before changing the code.

Both options 2 and 3 require immediate refactoring in FE, while the first one can be postponed (given that it's a very unlikely situation).

Reply:
We choosed option 3.